### PR TITLE
meson wrap for eigen 3.3.9

### PR DESCRIPTION
**Pull request by bitrunner**

### CODE REVIEW

File: upstream.wrap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[wrap-file]
+directory = eigen-3.3.9
+
+source_url = https://gitlab.com/libeigen/eigen/-/archive/3.3.9/eigen-3.3.9.tar.bz2
+source_filename = eigen-3.3.9.tar.bz2
+source_hash = 0fa5cafe78f66d2b501b43016858070d52ba47bd9b1016b0165a7b8e04675677
```
Comment (eli-schwartz) [inline upstream.wrap:6]: Upstream provides an `eigen3.pc` providing this dependency, usable via `dependency('eigen3')`.

Per https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section, please include:
```
[provide]
eigen3 = eigen_dep
```

This lets automatic fallback work, and we want "provide" sections for all new wraps.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+
+pkgc = import('pkgconfig')
+pkgc.generate(name: 'eigen3',
+    version: meson.project_version(),
+    description: 'A C++ template library for linear algebra: vectors, matrices, and related algorithms'
+)
+
+eigen_dep = declare_dependency(
+    include_directories : include_directories('.'),
+)
```
Comment (eli-schwartz) [inline meson.build]: This file seems to be missing its final newline.

File: upstream.wrap (hunk 1 above)
Comment (bitrunner) [reply upstream.wrap:6]: Copy.  Done.  Thanks for the quick and thorough feedback!

File: meson.build [hunk 3]
```diff
@@ -489,4 +489,8 @@ pkgc.generate(name: 'eigen3',
 
 eigen_dep = declare_dependency(
     include_directories : include_directories('.'),
-)
\ No newline at end of file
+)
+
+[provide]
+eigen3 = eigen_dep
+
```
Comment (eli-schwartz) [inline meson.build]: You seem to have accidentally added this to the wrong file. :p

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: meson.build (hunk 3 above)
Comment (bitrunner) [reply meson.build]: ugh... i'm not real smart...  thanks for sanity checking this carefully!

**Pull request merged**